lib/Sema/CSDiag.cpp

```diff
@@ -5470,22 +5469,9 @@ bool FailureDiagnosis::diagnoseCallContextualConversionErrors(
   if (withoutContextual.empty())
     return false;
 
-  SmallPtrSet<TypeBase *, 4> withContextual;
-  typeCheckExpr(TC, callExpr, DC, withContextual, contextualType);
-  // If type-checking with contextual type didn't produce any results
-  // it means that we have a contextual mismatch.
-  if (withContextual.empty()) {
```

Comment:
Sorry I'm a bit late to this, but it seems like this was an important condition - expression could produce a single result with and without contextual type so we wouldn't really want to diagnose this as contextual mismatch in cases like that (e.g. when re-typechecking a call with correct contextual but actual problem is that assignment is to r-value or something similar), it seems like we need more tests for situations like that...

Reply:
The patch that removes this was independent of everything else in this PR. I was at first hoping to rip out CalleeListener altogether but only got rid of one usage. If you can come up with a test case, feel free to revert this part.